File: front-end/src/app/domain/models/game/action/impl/jail/GoToJailAction.ts
import {GameAction} from "../../GameAction";
import {Player} from "../../../Player";

/**
 * TODO: Add JsDoc
 *
 * @author Hamza el Haouti
 */
export class GoToJailAction extends GameAction {

  public constructor(
    id: number,
    actor: Player,
    performedAt: Date,
  ) {
    super(GoToJailAction.name, id, actor, performedAt);
  }

  public static trueCopy(action: GoToJailAction): GoToJailAction | null {
    return new GoToJailAction(
      action.id,
      Player.trueCopy(action.actor),
      new Date(action.performedAt)
    );
  }

  public override toString(): string {
    return `${this.actor.user.username} has been jailed.`;
  }

}

File: back-end/src/main/java/nl/hva/backend/domain/models/game/action/impl/jail/GoToJailAction.java
package nl.hva.backend.domain.models.game.action.impl.jail;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import nl.hva.backend.domain.models.game.Player;
import nl.hva.backend.domain.models.game.action.GameAction;

import javax.persistence.Entity;

/**
 * TODO: Add JavaDoc
 *
 * @author Hamza el Haouti
 */
@Entity
@Getter
@Setter
@NoArgsConstructor
public class GoToJailAction extends GameAction {
    int toPostion;

    public GoToJailAction(
            Player actor,
            int toPostion
    ) {
        super(actor);
        this.toPostion = toPostion;
    }

}